HW_27.10/HW_27.10/Controllers/Controller.cs

```diff
@@ -0,0 +1,94 @@
+{
+    [ApiController]
+    [Route("book")]
+    public class Controller : ControllerBase
+    {
+        private readonly IConfiguration _config;
+        public Controller(IConfiguration config)
+        {
+            _config = config;
+        }
+
+        [HttpGet("all")]
+        public List<Book> GetAll()
+        {
+            using (StreamReader streamReader = new StreamReader(_config.GetValue<string>("Path1")))
+            {
+                var json = streamReader.ReadToEnd();
+                return JsonConvert.DeserializeObject<List<Book>>(json);
+            }
+        }
+
+        [HttpGet("one/{id}")]
+        public Book GetOne(Guid id)
+        {
+            using (StreamReader streamReader = new StreamReader(_config.GetValue<string>("Path1")))
+            {
+                var json = streamReader.ReadToEnd();
+                var books = JsonConvert.DeserializeObject<List<Book>>(json);
+                foreach (var item in books)
+                {
+                    if (id == item.Id)
+                    {
+                        return item;
+                    }
+                }
+                return null;
+            }
+        }
+
+        [HttpPost("add")]
+        public void Post([FromBody] Book book)
+        {
+            List<Book> books = new List<Book>();
+            var newBook = new Book() { Name = book.Name, Author = book.Author, Id = book.Id, Pages = book.Pages, Popularity = book.Popularity };
+
+            using (StreamReader streamReader = new StreamReader(_config.GetValue<string>("Path1")))
+            {
+                var json = streamReader.ReadToEnd();
+                books = JsonConvert.DeserializeObject<List<Book>>(json);
+            }
+
+            using (StreamWriter streamWriter = new StreamWriter(_config.GetValue<string>("Path1")))
+            {
+                books.Add(newBook);
+                var json = JsonConvert.SerializeObject(books, Formatting.Indented);
+                streamWriter.WriteLine(json);
+            }
+        }
+
+        [HttpDelete("delete/{id}")]
+        public void Delete(Guid id)
+        {
+            List<Book> books = new List<Book>();
+            using (StreamReader streamReader = new StreamReader(_config.GetValue<string>("Path1")))
+            {
+                var json = streamReader.ReadToEnd();
+                books = JsonConvert.DeserializeObject<List<Book>>(json);
+            }
+            using (StreamWriter streamWriter = new StreamWriter(_config.GetValue<string>("Path1")))
+            {
+               
+                for(int i=0; i<books.Count; i++)
+                {
+                    if (books[i].Id == id)
+                    {
+                        books.Remove(books[i]);
+                    }
+                }
+                var json = JsonConvert.SerializeObject(books, Formatting.Indented);
+                streamWriter.WriteLine(json);
+            }
+        }
+    }
+}
```

Comment:
you get _config.GetValue<string>("Path1") all the time
you could find better way to get this
for example get the path at ctor and store in a field or even put this to DI and use IOptions to get the value not controller construction
at least put "Path1" into const and use do not use the string every time